manager/integration/tests/test_settings.py

```diff
@@ -228,4 +231,98 @@ def test_setting_guaranteed_engine_cpu():
 
     Note: use fixture to restore the setting into the original state
     """
-    pass
+    client = get_longhorn_api_client()
+    core_api = get_core_api_client()
+
+    # Get current guaranteed engine cpu setting and save it to org_setting
+    setting = client.by_id_setting(SETTING_GUARANTEED_ENGINE_CPU)
+    org_setting = setting
+
+    # Get current running instance managers
+    instance_managers = client.list_instance_manager()
+
+    # Set an invalid value, and it should return error
+    try:
+        client.update(setting, value="xxx")
+    except Exception as e:
+        print("Exception when update "
+              "Guaranteed Engine CPU to xxx", e)
+        pass
+
+    # Update guaranteed engine cpu setting to 0.1
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.1",
+                                        "100m", instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0",
+                                        None, instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0.2
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.2",
```

Comment:
The input should be `200m` instead of `0.2`. We need to test different types of inputs.

Reply:
Oh, i see. I will update it.